app/routes/users/components/PasswordStrengthMeter.js

```diff
@@ -19,60 +17,76 @@ type Props = {
   user: UserEntity,
 };
 
-class PasswordStrengthMeter extends Component<Props> {
-  render() {
-    const { password, user } = this.props;
-    const zxcvbnValue = zxcvbn(
-      password,
-      //$FlowFixMe[incompatible-call]
-      Object.values(pick(user, ['username', 'firstName', 'lastName']))
-    );
-    let tips = zxcvbnValue.feedback.suggestions;
-    tips.push(zxcvbnValue.feedback.warning);
-    tips = tips.map((tip) => passwordFeedbackMessages[tip]).filter(Boolean);
+const PasswordStrengthMeter = ({ password, user }: Props) => {
+  const [zxcvbnValue, setZxcvbnResult] = useState();
 
-    const crackTimeSec = Number(
-      zxcvbnValue.crack_times_seconds.offline_slow_hashing_1e4_per_second
-    );
-    const crackTimeDuration = moment
-      .duration(crackTimeSec, 'seconds')
-      .humanize();
-    const crackTime =
-      crackTimeSec > 2 ? crackTimeDuration : crackTimeSec + ' sekunder';
+  useEffect(() => {
+    import('zxcvbn').then((zxcvbn) => {
+      setZxcvbnResult(
+        zxcvbn.default(password, [user.username, user.firstName, user.lastName])
+      );
+    });
+  }, [password, user.firstName, user.lastName, user.username]);
 
+  if (!zxcvbnValue) {
     return (
-      <Fragment>
-        <div className={styles.removeLabels}>
-          <Bar
-            labels={[1, 2, 3, 4, 5]}
-            labelColor="#000"
-            progress={zxcvbnValue.score * 25}
-            barColor={barColor[zxcvbnValue.score]}
-            seperatorColor="#fff"
-          />
-        </div>
-        {password && (
-          <Fragment>
-            <span>
-              <strong>Passordstyrke: </strong>{' '}
-              {passwordLabel[zxcvbnValue.score]}
-            </span>
-            <p>
-              Dette passordet hadde tatt en maskin {crackTime} å knekke @ 10^4
-              Hash/s.
-            </p>
-          </Fragment>
-        )}
-        {password && zxcvbnValue.score < 3 && (
-          <ul className={styles.tipsList}>
-            {tips.map((value, key) => {
-              return <li key={key}>{value}</li>;
-            })}
-          </ul>
-        )}
-      </Fragment>
+      <>
+        <PasswordStrengthBar strengthScore={0} />
+        <span>
+          <strong>Laster inn passordstyrke...</strong>
+        </span>
+      </>
     );
   }
-}
+
+  let tips = zxcvbnValue.feedback.suggestions;
+  tips.push(zxcvbnValue.feedback.warning);
+  tips = tips.map((tip) => passwordFeedbackMessages[tip]).filter(Boolean);
+
+  const crackTimeSec = Number(
+    zxcvbnValue.crack_times_seconds.offline_slow_hashing_1e4_per_second
```

Comment:
Here as well, can `crack_times_seconds` be undefined?

Reply:
I didn't actually change this code, so I assume it is working. But I guess it doesn't hurt to add some optional chaining and a default value in case it is undefined.